### PR TITLE
[Interpreter] Allow int32 inputs for BatchedReduceAdd

### PR DESCRIPTION
**Pull request by sgeorgiev255**
Summary:
This PR adds support for int32 inputs to the BatchedReduceAdd node.

Test Plan:
Tests have been added to InterpreterOperatorTest

### CODE REVIEW

File: lib/Backends/Interpreter/InterpreterNodes.cpp [hunk 1]
```diff
@@ -3716,7 +3734,14 @@ template <typename ElemTy>
...
     const ShapeVector &eDestDims) {
-  staticAssertFloatingPointType(ElemTy);
+  static_assert(
+      std::is_floating_point<ElemTy>::value ||
+          std::is_same<float16_t,
+                       typename std::remove_cv<ElemTy>::type>::value ||
+          std::is_same<bfloat16_t,
+                       typename std::remove_cv<ElemTy>::type>::value ||
+          std::is_same<int, typename std::remove_cv<ElemTy>::type>::value,
+      "This implementation is for floating-point and int32 values only");
```
Comment (jfix71) [inline lib/Backends/Interpreter/InterpreterNodes.cpp:3744]: It's a little funny it's called `FloatImpl`. Maybe we should rename to just `fwdBatchedReduceAddInstImpl`

**Review by jfix71 — APPROVED** [review]
Otherwise fine by me, thanks

File: lib/Backends/Interpreter/InterpreterNodes.cpp (hunk 1 above)
Comment (sgeorgiev255) [reply lib/Backends/Interpreter/InterpreterNodes.cpp]: Done.

**Comment by sgeorgiev255** [comment]
Hi @jfix71, if there are no more issue for this PR could you reopen it and merge it please?

**Comment by jfix71** [comment]
Sure, sorry for missing this 🙂

**Pull request closed without merging**